src/Fees.sol

```diff
@@ -1,58 +1,66 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.20;
 
+import {Log2} from "./Log2.sol";
+
 library PDPFees {
+    uint256 constant ATTO_FIL = 1;
+    uint256 constant ONE_FIL = 1e18 * ATTO_FIL;
 
-    // Estimate of USD price in FIL
-    uint256 constant FIL_USD_NUM = 10;
-    uint256 constant FIL_USD_DENOM = 35;
-           
-    uint256 constant ONE_PERCENT_DENOM = 100;
-
-    uint256 constant LEAVES_PER_TIB = 2^35;
-
-    uint256 constant EPOCHS_PER_MONTH = 86_400;
-    // Revenue in usd per TiB per month
-    uint256 constant MONTHLY_TIB_REVENUE_USD_NUM = 35;
-    uint256 constant MONTHLY_TIB_REVENUE_USD_DENOM = 10;
-
-    uint256 constant PROOF_GAS_FLOOR = 2_000_000;
-    uint256 constant ONE_NANO_FIL = 10^9;
-    uint256 constant ONE_FIL = 10^18;
-
-    uint256 constant SYBIL_FEE = 10^17;
-
-    // Currently unused
-    //
-    // Returns the service fee for a given proofset duration, and size
-    // Size measured in leaves (32 byte chunks), duration measured in epochs
-    function serviceFee(uint256 duration, uint256 size) internal pure returns (uint256) {
-        // 1 FIL / TiB / month ~ revenue 
-        // 86,400 epochs / month
-        // 2^35 leafs / TiB
-        // fee is 1% of revenue
-
-        // [(Monthly revenue in USD) * (FIL / USD) * size * duration] / 
-        // [Leaves per TiB * Epochs per month * 1%]
-        uint256 numerator = ONE_FIL * MONTHLY_TIB_REVENUE_USD_NUM * FIL_USD_NUM * size * duration;
-        uint256 denominator = MONTHLY_TIB_REVENUE_USD_DENOM * FIL_USD_DENOM * LEAVES_PER_TIB * EPOCHS_PER_MONTH * ONE_PERCENT_DENOM;
-        return numerator / denominator;
-    }
+    // 0.1 FIL
+    uint256 constant SYBIL_FEE = ONE_FIL / 10;
+
+    // assume 1FIL = $5 for now -> we can change this to use an oracle in PDP V1
+    uint256 constant FIL_USD_PRICE = 5;
```

Comment:
@ZenGround0 

See discussion [here](https://www.notion.so/Pricing-mechanism-for-PDPverifier-12adc41950c180ea9608cb419c369ba4?d=151dc41950c180889134001c2a9d89b6&pvs=4#145dc41950c180a78fb5ee20ef9958ca)

1) We can either ship this now with this price for PDP V0 and revisit in PDP V1 etc once the on chain payments/FWS story becomes clearer

2) Change this to use an oracle right now that gives us a real time price feed and live with the complexity.

What are your thoughts ?

Reply:
recaping here that we decided to move forward implementing a call to a price oracle already in v0!

Reply:
[not relevant since we are going for the price oracle, just putting this here for clarity: in the current case, if the fil price is higher than what is set up here, the fee is going to be higher than expected (i.e. if FIL gets to 10 and price is set to 5 in the contract, fee would be 2x the expected value)]

Reply:
Have you considered using a stablecoin as payment for PDP? We'd love to work with you to figure out how to make this happen!

Reply:
@usdfil we plan build a payment pathway with fungible token support - but that will be the next milestone & will be outside of the proof contract itself (out of the scope of this PR). It is better to follow #fil-pdp, and we would like to get your support when we start working on payments.